DESIGN.md

```diff
@@ -0,0 +1,36 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request.
+* Support usage of requests to URLs directed to either `http` or `https`.
+* Support the supplying of request headers.
+* Expose response headers returned from request.
+* Expose response body as string where applicable.
+
+## Patterns
+
+Specific to this provider:
+
+* Only idempotent GET requests are supported.
+* Only 200 status codes are considered successful.
```

Comment:
And I assume this will be updated once your RFC is finalised and work begin to expand this bit?